### ISSUE TITLE
[azure-core] Raise better error message for malformed endpoint

### ISSUE THREAD

**Issue by kristapratico** [Azure/azure-sdk-for-python#11048]
In reference to this issue: #9521 

If the endpoint passed is malformed in some way, e.g. the https:// is chopped off, we get a cryptic message raised from azure.core :

```  
File "C:\Users\krpratic\azure-sdk-for-python\env\lib\site-packages\azure\core\pipeline\transport\_base.py", line 695, in format_url
    base = self._base_url.format(**kwargs).rstrip("/")
KeyError: 'Endpoint'
```

KeyError should be raised with a better error message, e.g., like:

> ‘Failed to make service call. Did not understand the endpoint passed in: {some broken endpoint}. Please ensure that it is a valid URL’.

**Comment by lmazuel** [comment]
Full stacktrace:
```python
Traceback (most recent call last):
  File ".\broken_endpoint.py", line 15, in <module>
    response = text_analytics_client.analyze_sentiment(documents, language="en")
  File "c:\users\lmazuel\git\azure-sdk-for-python\sdk\core\azure-core\azure\core\tracing\decorator.py", line 83, in wrapper_use_tracer
    return func(*args, **kwargs)
  File "c:\users\lmazuel\git\azure-sdk-for-python\sdk\textanalytics\azure-ai-textanalytics\azure\ai\textanalytics\_text_analytics_client.py", line 414, in analyze_sentiment
    **kwargs
  File "c:\users\lmazuel\git\azure-sdk-for-python\sdk\textanalytics\azure-ai-textanalytics\azure\ai\textanalytics\_generated\operations\_text_analytics_client_operations.py", line 373, in sentiment
    request = self._client.post(url, query_parameters, header_parameters, **body_content_kwargs)
  File "c:\users\lmazuel\git\azure-sdk-for-python\sdk\core\azure-core\azure\core\pipeline\transport\_base.py", line 798, in post
    "POST", url, params, headers, content, form_content, stream_content
  File "c:\users\lmazuel\git\azure-sdk-for-python\sdk\core\azure-core\azure\core\pipeline\transport\_base.py", line 685, in _request
    request = HttpRequest(method, self.format_url(url))
  File "c:\users\lmazuel\git\azure-sdk-for-python\sdk\core\azure-core\azure\core\pipeline\transport\_base.py", line 721, in format_url
    base = self._base_url.format(**kwargs).rstrip("/")
KeyError: 'Endpoint'
```

**Comment by xiangyan99** [comment]
Fixed in #12106
